# Gastos do Município de Capelinha/MG no Ano de 2021

![Capelinha, Estado de Minas Gerais!](capelinha_mg_overview.jpg "Capelinha/MG")

## Município de Capelinha/MG

[Capelinha](https://pt.wikipedia.org/wiki/Capelinha) é um município brasileiro no interior do estado de Minas Gerais, Região Sudeste do país. Localiza-se no Vale do Jequitinhonha e sua população em julho de 2020 foi estimada em 38.057 habitantes. Exerce uma polarização econômica, educacional, cultural, empresarial e esportiva sobre os municípios próximos, constituindo a área mais ocupada demograficamente do Vale do Jequitinhonha. A povoação se iniciou em 1812, a partir da construção da Igreja Matriz de Nossa Senhora da Graça por Feliciano Luiz Pego.


> "Sem dados você é apenas mais uma pessoa com uma opinião." - W. Edwards Deming

## Índice

1. [Dados Abertos](#dados-abertos)
1. [Limpeza de Dados (Data Cleaning)](#limpeza-de-dados-data-cleaning)
2. [Análise dos Dados (Data Analysis)](#análise-dos-dados-data-analysis)
3. [Visualização dos Dados (Data Visualization)](#visualização-dos-dados-data-visualization)
4. [Conclusão](#conclusão)

# Dados Abertos

## Conceito
Segundo a [Open Knowledge Foundation no Brasil - OKBR](https://ok.org.br), _“dados são abertos quando qualquer pessoa pode livremente *usá-los*, *reutilizá-los* e *redistribuí-los*, estando sujeito a, no máximo, a exigência de creditar a sua autoria e compartilhar pela mesma licença”_. Quando os dados são produzidos, coletados ou custodiados por autoridades públicas e disponibilizados em formato aberto, nesse caso temos dados abertos governamentais. Ainda segundo a OKBR, dados abertos também são pautados por três leis e oito princípios:

As três leis:

- I - Se o dado não pode ser encontrado e indexado na Web, ele não existe;

- II - Se não estiver aberto e disponível em formato compreensível por máquina, ele não pode ser reaproveitado; e

- III - Se algum dispositivo legal não permitir sua replicação, ele não é útil.

Os oito princípios:

1) *Completos*: todos os dados públicos são disponibilizados. Dados são informações eletronicamente gravadas, incluindo, mas não se limitando a, documentos, bancos de dados, transcrições e gravações audiovisuais. Dados públicos são dados que não estão sujeitos a limitações válidas de privacidade, segurança ou controle de acesso, reguladas por estatutos.

2) *Primários*: os dados são publicados na forma coletada na fonte, com a mais fina granularidade possível, e não de forma agregada ou transformada.

3) *Atuais*: os dados são disponibilizados o quão rapidamente seja necessário para preservar o seu valor.

4) *Acessíveis*: os dados são disponibilizados para o público mais amplo possível e para os propósitos mais variados possíveis.

5) *Processáveis por máquina*: os dados são razoavelmente estruturados para possibilitar o seu processamento automatizado.

6) *Acesso não discriminatório*: os dados estão disponíveis a todos, sem que seja necessária identificação ou registro.

7) *Formatos não proprietários*: os dados estão disponíveis em um formato sobre o qual nenhum ente tenha controle exclusivo.

8) *Livres de licenças*: os dados não estão sujeitos a regulações de direitos autorais, marcas, patentes ou segredo industrial. Restrições razoáveis de privacidade, segurança e controle de acesso podem ser permitidas na forma regulada por estatutos.

O objetivo, além da promoção da transparência governamental, é fornecer insumos aos cidadãos e à sociedade civil para serem desenvolvidas ferramentas que proporcionem uma percepção mais efetiva das atividades e dos gastos do governo e contribuam para a melhoria da gestão pública, o estímulo ao controle e participação social, a geração de emprego e renda e o fomento à inovação tecnológica.

No Brasil, conforme o [Manual dos Dados Abertos: Governo](https://www.w3c.br/pub/Materiais/PublicacoesW3C/Manual_Dados_Abertos_WEB.pdf), dados abertos governamentais são particularmente importantes devido à quantidade e centralidade dos dados que coleta e, também, por tais dados serem públicos, um direito garantido no art. 5º da Constituição Federal brasileira.

## Legislação

Abaixo alguns dos dispositivos legais que determinam, direta ou indiretamente, a disponibilização de dados abertos para participação da sociedade no debate das finanças públicas:

- *Lei Complementar n.º 101/2000 (Lei de Responsabilidade Fiscal - LRF)*: determina ao Poder Público a adoção de instrumentos de transparência na gestão fiscal em meios eletrônicos de acesso público às informações orçamentárias e prestações de contas. Disponível em: [https://www.planalto.gov.br/ccivil_03/leis/lcp/lcp101.htm](https://www.planalto.gov.br/ccivil_03/leis/lcp/lcp101.htm)

- *Lei Complementar n.º 131/2009 (Lei da Transparência)*: acrescenta dispositivos à Lei Complementar nº 101, de 4 de maio de 2000, que estabelece normas de finanças públicas voltadas para a responsabilidade na gestão fiscal e dá outras providências, a fim de determinar a disponibilização, em tempo real, de informações pormenorizadas sobre a execução orçamentária e financeira da União, dos Estados, do Distrito Federal e dos Municípios. Disponível em: [https://www.planalto.gov.br/ccivil_03/leis/LCP/Lcp131.htm](https://www.planalto.gov.br/ccivil_03/leis/LCP/Lcp131.htm)

- *Lei n.º 12.257/2012 (Lei de Acesso à Informação)*: dispõe sobre os procedimentos a serem observados pela União, Estados, Distrito Federal e Municípios, com o fim de garantir o acesso a informações previsto no art. 5º, no inciso II do § 3º do art. 37 e no § 2º do art. 216 da Constituição Federal. Disponível em: [https://www.planalto.gov.br/ccivil_03/_ato2011-2014/2011/lei/l12527.htm](https://www.planalto.gov.br/ccivil_03/_ato2011-2014/2011/lei/l12527.htm)

- *Decreto n.º 7.724 de 16 de maio de 2012*: Regulamenta a Lei no 12.527 (Lei de Acesso à Informação), que dispõe sobre o acesso a informações previsto no inciso XXXIII do caput do art. 5º, no inciso II do § 3º do art. 37 e no § 2º do art. 216 da Constituição Federal. Disponível em: [https://www.planalto.gov.br/ccivil_03/_ato2011-2014/2012/decreto/d7724.htm](https://www.planalto.gov.br/ccivil_03/_ato2011-2014/2012/decreto/d7724.htm)

- *Decreto n.º 10.160 de 09 de dezembro de 2019*: Institui a Política Nacional de Governo Aberto e o Comitê Interministerial de Governo Aberto. Disponível em: [https://www.planalto.gov.br/ccivil_03/_Ato2019-2022/2019/Decreto/D10160.htm](https://www.planalto.gov.br/ccivil_03/_Ato2019-2022/2019/Decreto/D10160.htm)

- *Instrução Normativa INDA n.º 4/2012*: institui a Infraestrutura Nacional de Dados Abertos (Inda). Disponível em: [https://dados.gov.br/pagina/instrucao-normativa-da-inda](https://dados.gov.br/pagina/instrucao-normativa-da-inda)

# Disclaimer

O storytelling deste estudo foi baseada na proposta de permitir a qualquer pessoa conseguir compreender os dados exibidos e o 'modus operandi' utilizado em sua elaboração. Caso você já tenha conhecimento técnico em Python e/ou Data Science, desconsidere comentários qualquer cometário enfadonho.

# Limpeza de Dados (Data Cleaning)

Que a planilha contém muitos dados, isso é fato. Para podermos ter um melhor aproveitamento das informações disponibilizadas, precisamos realizar a conversão de alguns dados que foram importados incorretamente para o script. Abaixo, você poderá visualizar o código que escrevi, assim como a explicação de todo o processo.

## Importando bibliotecas

In [ ]:
# Caso seja necessário instalar as bibliotecas externas utilizadas nesse estudo execute esta célula.
%pip install pandas
%pip install numpy
%pip install matplotlib
%pip install seaborn

In [2]:
# Para criarmos os DataFrames.
# https://pandas.pydata.org
import pandas as pd

# Para realizarmos cálculos matemáticos complexos.
# https://numpy.org
import numpy as np

# Para podermos visualizar os dados em gráficos.
# https://matplotlib.org
import matplotlib.pyplot as plt

# Permite deixarmos a aparência dos gráficos mais atraentes.
# https://seaborn.pydata.org
import seaborn as sns

In [3]:
# Configurando o Pandas para exibir os valores de ponto flutuante
# utilizando duas casas decimais apos a virgula.
pd.options.display.float_format = '{:,.2f}'.format

In [4]:
# Arquivo contendo os dados públicos do Município de Capelinha/MG para o ano de 2021.
# Os dados foram obtidos no Portal de Dados Abertos do Tribunal de Contas do Estado de Minas Gerais.
# https://dadosabertos.tce.mg.gov.br

# Informações referentes as Despesas do município.
FONTE_DESPESAS_2021 = r'data\2021_despesa.csv'

#  Informações referentes as Liquidações realizadas pelo município.
FONTE_PAGAMENTOS_2021 = r'data\2021_pagamento.csv'

In [5]:
# Vamos começar carregando os dados presentes nos arquivos CSV.
# O delimitador padrão dos dados é semicolon ;
# A codificação dos dados no arquivo é UTF-8

despesas_2021 = pd.read_csv(FONTE_DESPESAS_2021, sep=';', encoding='utf-8')

pagamentos_2021 = pd.read_csv(FONTE_PAGAMENTOS_2021, sep=';', encoding='utf-8')

## Entendo a estrutura dos nossos dataset

Para melhor visualização dos dados que estão contidos na planilha, abaixo podemos conferir a estrutura dos dataframes:

In [6]:
despesas_2021.sample(20)

seq_orgao  cod_municipio  seq_unidade  cod_unidade  cod_subunidade  \
7534         372        3112307       390320         4001               1   
476          372        3112307       390322         8001               3   
8223         372        3112307       390332        11001               1   
3877         372        3112307       390320         4001               1   
4945         372        3112307       390347         9002               3   
5816         372        3112307       390337         7001               2   
10955        372        3112307       390312         5001               3   
2202         372        3112307       390320         4001               1   
4194         372        3112307       390337         7001               2   
6432         372        3112307       390315        10002               1   
8750         372        3112307       390325         8001               4   
3278         371        3112307       390307         1001               2   
1082         372        3112307       390332        11001               1   
3629         372        3112307       390330        13001               3   
7964         372        3112307       390338         7001               1   
6381         372        3112307       390324         8001               2   
10630        372        3112307       390350         9002               4   
7224         371        3112307       390307         1001               2   
9650         372        3112307       390316         9001               1   
3312         372        3112307       390316         9001               1   

       num_anoexercicio  num_mesexercicio                  dsc_funcao  \
7534               2021                12             02 - JUDICIÁRIA   
476                2021                12                  10 - SAÚDE   
8223               2021                11            20 - AGRICULTURA   
3877               2021                 9             02 - JUDICIÁRIA   
4945               2021                 5     08 - ASSISTÊNCIA SOCIAL   
5816               2021                12               12 - EDUCAÇÃO   
10955              2021                 9           24 - COMUNICAÇÕES   
2202               2021                 3             02 - JUDICIÁRIA   
4194               2021                 8               12 - EDUCAÇÃO   
6432               2021                 1                13 - CULTURA   
8750               2021                12                  10 - SAÚDE   
3278               2021                 3            01 - LEGISLATIVA   
1082               2021                10            20 - AGRICULTURA   
3629               2021                 6              15 - URBANISMO   
7964               2021                 7               12 - EDUCAÇÃO   
6381               2021                 3                  10 - SAÚDE   
10630              2021                12     08 - ASSISTÊNCIA SOCIAL   
7224               2021                10            01 - LEGISLATIVA   
9650               2021                 5     08 - ASSISTÊNCIA SOCIAL   
3312               2021                 1  14 - DIREITOS DA CIDADANIA   

                                           dsc_subfuncao  \
7534   062 - DEFESA DO INTERESSE PÚBLICO NO PROCESSO ...   
476          302 - ASSISTÊNCIA HOSPITALAR E AMBULATORIAL   
8223                           122 - ADMINISTRAÇÃO GERAL   
3877   062 - DEFESA DO INTERESSE PÚBLICO NO PROCESSO ...   
4945                       244 - ASSISTÊNCIA COMUNITÁRIA   
5816                            361 - ENSINO FUNDAMENTAL   
10955                           131 - COMUNICAÇÃO SOCIAL   
2202   062 - DEFESA DO INTERESSE PÚBLICO NO PROCESSO ...   
4194                            361 - ENSINO FUNDAMENTAL   
6432                              392 - DIFUSÃO CULTURAL   
8750                          304 - VIGILÂNCIA SANITÁRIA   
3278                           122 - ADMINISTRAÇÃO GERAL   
1082                           122 - ADMINISTRAÇÃO GERAL   
3629           

In [7]:
# Visualizando as dimensões da nossa base de dados.
print(f'Este DataFrame possui {despesas_2021.shape[0]} linhas e {despesas_2021.shape[1]} colunas.')

Este DataFrame possui 11178 linhas e 23 colunas.


In [8]:
despesas_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11178 entries, 0 to 11177
Data columns (total 23 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   seq_orgao            11178 non-null  int64  
 1   cod_municipio        11178 non-null  int64  
 2   seq_unidade          11178 non-null  int64  
 3   cod_unidade          11178 non-null  int64  
 4   cod_subunidade       11178 non-null  int64  
 5   num_anoexercicio     11178 non-null  int64  
 6   num_mesexercicio     11178 non-null  int64  
 7   dsc_funcao           11178 non-null  object 
 8   dsc_subfuncao        11178 non-null  object 
 9   dsc_programa         11178 non-null  object 
 10  dsc_acao             11178 non-null  object 
 11  dsc_subacao          0 non-null      float64
 12  dsc_naturezadespesa  11178 non-null  object 
 13  dsc_fonterecurso     11178 non-null  object 
 14  vlr_previsto         11178 non-null  float64
 15  vlr_acrescimo        11178 non-null 

Nosso dataframe, contendo as informações referentes as _Despesas_ pelo município, é composto por 11.178 linhas e 23 colunas, com as seguintes informações:

- seq_orgao: Sequencial identificador do órgão.
- cod_municipio: Código IBGE do município.
- seq_unidade: Sequencial identificador da unidade orçamentária. 
- cod_unidade: Unidades orçamentárias do município
- cod_subunidade: Subunidades orçamentárias do município. Recebe -9 caso o valor seja nulo.
- num_anoexercicio: Número do ano de referência da informação.
- num_mesexercicio: Número do mês de referência da informação.
- dsc_funcao: Representa a função na qual foi destinada a previsão / execução da despesa.
- dsc_subfuncao: Representa a subfuncão na qual foi destinada a previsão / execução da despesa.
- dsc_programa: Representa o programa no qual foi destinada a previsão / execução da despesa.
- dsc_acao: Representa a ação na qual foi destinada a previsão / execução da despesa.
- dsc_subacao: Representa a subação na qual foi destinada a previsão / execução da despesa.
- dsc_naturezadespesa: Representa a natureza da despesa na qual foi destinada a previsão / execução da despesa.
- dsc_fonterecurso: Representa a fonte de recurso na qual foi destinada a previsão / execução da despesa
- vlr_previsto: Valor previsto das despesas informadas pelo município. 
- vlr_acrescimo: Valor acrescido das despesas previstas inicialmente informadas pelo município.
- vlr_reducao: Valor das deduções das despesas previstas inicialmente informadas pelo município.
- vlr_empenhado: Valor empenhado das despesas atualizadas informadas pelo município.
- vlr_liquidado: Valor liquidado das despesas atualizadas informadas pelo município.
- vlr_pago: Valor pago das despesas atualizadas informadas pelo município.
- vlr_rspprocessado: Valor liquidado que não foi pago no exercício das despesas atualizadas informadas pelo município.
- vlr_rspnprocprocessado: Valor empenhado que não foi liquidado no exercício das despesas atualizadas informadas pelo município.
- num_versaoarq: Corresponde quanto a versão do arquivo.

In [9]:
pagamentos_2021.sample(20)

seq_pagamento  seq_orgao  seq_unidade  seq_empenho  seq_rsp  \
2958       114411730        372       390343     73821068       -1   
11582      120039600        372       390349     77450062       -1   
3801       114411221        372       390324     71953099       -1   
4384       116309134        372       390346     73821199       -1   
1666       126993711        372       390322     82605290       -1   
9258       112573552        372       390324     71952967       -1   
13455      122327210        372       390316     78454599       -1   
5327       116307973        372       390322     71951095       -1   
13177      125838690        372       390327     82002859       -1   
5046       109653329        372       390322     71951505       -1   
4462       116309171        372       390338     74956395       -1   
7918       109651794        372       390322     71951033       -1   
8052       109653126        372       390322     71950913       -1   
17115      118428245        372       390326     71951352       -1   
3739       126992419        372       390321     74956599       -1   
12280      120039739        372       390337     78454170       -1   
17602      118427799        372       390321     74956761       -1   
17137      118428882        372       390322     76167081       -1   
370        127346054        371       390307     78955444       -1   
17683      118427803        372       390331     71950880       -1   

       seq_liquidacao  seq_orgao_empenho  seq_unid_empenho  \
2958        121561827                372            390343   
11582       127190700                372            390349   
3801        121561457                372            390324   
4384        123476454                372            390346   
1666        134127102                372            390322   
9258        116849421                372            390324   
13455       127191847                372            390316   
5327        121561981                372            390322   
13177       133031377                372            390327   
5046        116847019                372            390322   
4462        121561089                372            390338   
7918        116846086                372            390322   
8052        116847966                372            390322   
17115       123477531                372            390326   
3739        134127657                372            390321   
12280       127190858                372            390337   
17602       121561357                372            390321   
17137       125584707                372            390322   
370         130416170                371            390307   
17683       125586145                372            390331   

       num_ano_referencia  num_mes_referencia  ... num_ord_pagamento  \
2958                 2021                   5  ...              6303   
11582                2021                   8  ...             11115   
3801                 2021                   5  ...              6754   
4384                 2021                   6  ...              8193   
1666                 2021                  12  ...             19979   
9258                 2021                   4  ...              3711   
13455                2021                   9  ...             11432   
5327                 2021                   6  ...              7412   
13177                2021                  11  ...             16855   
5046                 2021                   3  ...              2175   
4462                 2021                   6  ...              7306   
7918                 2021                   1  ...               431   
8052                 2021                   2  ...              1412   
17115                2021                   7  ...              9817   
3739                 2021                  12  ...             18558   
12280                2021                   8  ...             12139   
17602    

In [10]:
# Visualizando as dimensões da nossa base de dados.
print(f'Este DataFrame possui {pagamentos_2021.shape[0]} linhas e {pagamentos_2021.shape[1]} colunas.')

Este DataFrame possui 18383 linhas e 28 colunas.


In [11]:
pagamentos_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18383 entries, 0 to 18382
Data columns (total 28 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   seq_pagamento       18383 non-null  int64  
 1   seq_orgao           18383 non-null  int64  
 2   seq_unidade         18383 non-null  int64  
 3   seq_empenho         18383 non-null  int64  
 4   seq_rsp             18383 non-null  int64  
 5   seq_liquidacao      18383 non-null  int64  
 6   seq_orgao_empenho   18383 non-null  int64  
 7   seq_unid_empenho    18383 non-null  int64  
 8   num_ano_referencia  18383 non-null  int64  
 9   num_mes_referencia  18383 non-null  int64  
 10  num_doc_resp        18383 non-null  object 
 11  nom_resp            18383 non-null  object 
 12  num_doc_credor      18383 non-null  object 
 13  nom_credor          18383 non-null  object 
 14  num_empenho         18383 non-null  int64  
 15  dat_empenho         18383 non-null  object 
 16  num_

Nosso dataframe, contendo as informações referentes aos _Pagamentos_ realizadas pelo município, é composto por 18383 linhas e 28 colunas, com as seguintes informações:

- seq_pagamento: Sequencial identificador do pagamento.
- seq_orgao: Sequencial identificador do órgão.
- seq_unidade: Sequencial identificador da unidade orçamentário.
- seq_empenho: Sequencial identificador do empenho.
- seq_rsp: Sequencial identificador do restos a pagar.
- seq_liquidacao: Sequencial identificador da liquidação.
- seq_orgao_empenho: Sequencial identificador do órgão responsável pelo empenho.
- seq_unid_empenho: Sequencial identificador da unidade orçamentária do empenho.
- num_ano_referencia: Representa o ano de referência da informação.
- num_mes_referencia: Representa o mês de referência da informação.
- num_doc_resp: Número do documento do responsável pelo pagamento.
- nom_resp: Nome do responsável pelo pagamento.
- num_doc_credor: Número de documento do credor do pagamento.
- nom_credor: Nome do credor do pagamento.
- num_empenho: Número do empenho.
- dat_empenho: Data do empenho.
- num_liquidacao: Número da liquidação.
- dat_liquidacao: Data da liquidação.
- num_ord_pagamento: Número da ordem de pagamento.
- dat_pagamento: Data do pagamento.
- dsc_pagamento: Descrição do pagamento.
- dsc_tipo_pagamento: Descrição do tipo do pagamento (1 – Pagamento de despesa, exceto Serviço da Dívida; 2 – Pagamento de Serviço da Dívida; 3 – Pagamento de Restos a Pagar Processados e Não Processados Liquidados em Exercícios Anteriores; 4 – Pagamento de Restos a Pagar Não Processados Liquidados no Exercício Atual; 5 – Pagamento de despesa empenhada por outro órgão; 6 – Pagamento realizado por outro órgão)
- dsc_fonte_recurso: Descrição da fonte de recurso.
- vlr_pag_fonte: Valor pago por fonte de recurso.
- vlr_ret_fonte: Valor retido do pagamento por fonte de recurso.
- vlr_ant_fonte: Valor antecipado do pagamento por fonte de recurso.
- vlr_anu_fonte: Valor anulado do pagamento por fonte de recurso.
- num_versaoarq: Corresponde quanto a versão do arquivo

## Procurando por valores em branco

In [12]:
# Existe algum valor ausente no dataframe?
despesas_2021.isnull().values.any()

True

In [13]:
# Existe algum valor ausente no dataframe?
pagamentos_2021.isnull().values.any()

True

Ambos os dataframes apresentam valores ausentes, precisamos olhar mais de perto para verificar se essas informações ausentes podem prejudicar o processo de análise dos gastos do município.

In [14]:
# Visualizando os valores ausentes das informações referentes as despesas. 
despesas_2021.isnull().head(10)

seq_orgao  cod_municipio  seq_unidade  cod_unidade  cod_subunidade  \
0      False          False        False        False           False   
1      False          False        False        False           False   
2      False          False        False        False           False   
3      False          False        False        False           False   
4      False          False        False        False           False   
5      False          False        False        False           False   
6      False          False        False        False           False   
7      False          False        False        False           False   
8      False          False        False        False           False   
9      False          False        False        False           False   

   num_anoexercicio  num_mesexercicio  dsc_funcao  dsc_subfuncao  \
0             False             False       False          False   
1             False             False       False          False   
2             False             False       False          False   
3             False             False       False          False   
4             False             False       False          False   
5             False             False       False          False   
6             False             False       False          False   
7             False             False       False          False   
8             False             False       False          False   
9             False             False       False          False   

   dsc_programa  ...  dsc_fonterecurso  vlr_previsto  vlr_acrescimo  \
0         False  ...             False         False          False   
1         False  ...             False         False          False   
2         False  ...             False         False          False   
3         False  ...             False         False          False   
4         False  ...             False         False          False   
5         False  ...             False         False          False   
6         False  ...             False         False          False   
7         False  ...             False         False          False   
8         False  ...             False         False          False   
9         False  ...             False         False          False   

   vlr_deducao  vlr_empenhado  vlr_liquidado  vlr_pago  vlr_rspprocessado  \
0        False          False          False     False              False   
1        False          False          False     False              False   
2        False          False          False     False              False   
3        False          False          False     False              False   
4        False          False          False     False              False   
5        False          False          False     False              False   
6        False          False          False     False              False   
7        False          False          False     False              False   
8        False          False          False     False              False   
9        False          False          False     False              False   

   vlr_rspnprocessado  num_versao_arq  
0               False           False  
1               False           False  
2               False           False  
3               False           False  
4               False           False  
5               False           False  
6               False           False  
7               False           False  
8               False           False  
9               False           False  

[10 rows x 23 columns]

In [15]:
# Visualizando os valores ausentes das informações referentes as liquidações.
pagamentos_2021.isnull().head(10)

seq_pagamento  seq_orgao  seq_unidade  seq_empenho  seq_rsp  \
0          False      False        False        False    False   
1          False      False        False        False    False   
2          False      False        False        False    False   
3          False      False        False        False    False   
4          False      False        False        False    False   
5          False      False        False        False    False   
6          False      False        False        False    False   
7          False      False        False        False    False   
8          False      False        False        False    False   
9          False      False        False        False    False   

   seq_liquidacao  seq_orgao_empenho  seq_unid_empenho  num_ano_referencia  \
0           False              False             False               False   
1           False              False             False               False   
2           False              False             False               False   
3           False              False             False               False   
4           False              False             False               False   
5           False              False             False               False   
6           False              False             False               False   
7           False              False             False               False   
8           False              False             False               False   
9           False              False             False               False   

   num_mes_referencia  ...  num_ord_pagamento  dat_pagamento  dsc_pagamento  \
0               False  ...              False          False          False   
1               False  ...              False          False          False   
2               False  ...              False          False          False   
3               False  ...              False          False          False   
4               False  ...              False          False          False   
5               False  ...              False          False          False   
6               False  ...              False          False          False   
7               False  ...              False          False          False   
8               False  ...              False          False          False   
9               False  ...              False          False          False   

   dsc_tipo_pagamento  dsc_fonte_recurso  vlr_pag_fonte  vlr_ret_fonte  \
0               False              False          False          False   
1               False              False          False          False   
2               False              False          False          False   
3               False              False          False          False   
4               False              False          False          False   
5               False              False          False          False   
6               False              False          False          False   
7               False              False          False          False   
8               False              False          False          False   
9               False              False          False          False   

   vlr_ant_fonte  vlr_anu_fonte  num_versao_arq  
0          False          False           False  
1          False          False           False  
2          False          False           False  
3          False          False           False  
4          False          False           False  
5          False          False           False  
6          False          False           False  
7          False          False           False  
8          False          False           False  
9          False          False           False  

[10 rows x 28 columns]

In [16]:
# Verificando onde estão estes valores ausentes.
despesas_2021.isnull().sum()

seq_orgao                  0
cod_municipio              0
seq_unidade                0
cod_unidade                0
cod_subunidade             0
num_anoexercicio           0
num_mesexercicio           0
dsc_funcao                 0
dsc_subfuncao              0
dsc_programa               0
dsc_acao                   0
dsc_subacao            11178
dsc_naturezadespesa        0
dsc_fonterecurso           0
vlr_previsto               0
vlr_acrescimo              0
vlr_deducao                0
vlr_empenhado              0
vlr_liquidado              0
vlr_pago                   0
vlr_rspprocessado          0
vlr_rspnprocessado         0
num_versao_arq             0
dtype: int64

In [17]:
# Verificando onde estão estes valores ausentes.
pagamentos_2021.isnull().sum()

seq_pagamento           0
seq_orgao               0
seq_unidade             0
seq_empenho             0
seq_rsp                 0
seq_liquidacao          0
seq_orgao_empenho       0
seq_unid_empenho        0
num_ano_referencia      0
num_mes_referencia      0
num_doc_resp            0
nom_resp                0
num_doc_credor          0
nom_credor              0
num_empenho             0
dat_empenho             0
num_liquidacao        361
dat_liquidacao        361
num_ord_pagamento       0
dat_pagamento           0
dsc_pagamento           0
dsc_tipo_pagamento      0
dsc_fonte_recurso       0
vlr_pag_fonte           0
vlr_ret_fonte           0
vlr_ant_fonte           0
vlr_anu_fonte           0
num_versao_arq          0
dtype: int64

In [18]:
# Visualizando os valores ausentes em percentagem.
despesas_2021.isna().mean()

seq_orgao             0.00
cod_municipio         0.00
seq_unidade           0.00
cod_unidade           0.00
cod_subunidade        0.00
num_anoexercicio      0.00
num_mesexercicio      0.00
dsc_funcao            0.00
dsc_subfuncao         0.00
dsc_programa          0.00
dsc_acao              0.00
dsc_subacao           1.00
dsc_naturezadespesa   0.00
dsc_fonterecurso      0.00
vlr_previsto          0.00
vlr_acrescimo         0.00
vlr_deducao           0.00
vlr_empenhado         0.00
vlr_liquidado         0.00
vlr_pago              0.00
vlr_rspprocessado     0.00
vlr_rspnprocessado    0.00
num_versao_arq        0.00
dtype: float64

In [19]:
# Visualizando os valores ausentes em percentagem.
pagamentos_2021.isna().mean()

seq_pagamento        0.00
seq_orgao            0.00
seq_unidade          0.00
seq_empenho          0.00
seq_rsp              0.00
seq_liquidacao       0.00
seq_orgao_empenho    0.00
seq_unid_empenho     0.00
num_ano_referencia   0.00
num_mes_referencia   0.00
num_doc_resp         0.00
nom_resp             0.00
num_doc_credor       0.00
nom_credor           0.00
num_empenho          0.00
dat_empenho          0.00
num_liquidacao       0.02
dat_liquidacao       0.02
num_ord_pagamento    0.00
dat_pagamento        0.00
dsc_pagamento        0.00
dsc_tipo_pagamento   0.00
dsc_fonte_recurso    0.00
vlr_pag_fonte        0.00
vlr_ret_fonte        0.00
vlr_ant_fonte        0.00
vlr_anu_fonte        0.00
num_versao_arq       0.00
dtype: float64

A base da dados fornecida pelo Tribunal de Contas do Estado de Minas Gerais é bem completa e praticamente não apresenta informações faltantes, facilitando em muito o trabalho de análise dos dados.

## Eliminando informações irrelevantes

Algumas das informações presentes nos dataframes são irrelevantes para o objetivo da análise em curso, portanto, vamos eliminá-las facilitando, assim o trabalho com os dados.

In [20]:
# Excluindo colunas desnecessárias do dataframe.
# axis especifica se os dados serão descartados do índice/linha (0 ou index) ou coluna (1 ou column).
# inplace=True indica que os dados serão alterados diretamente no DataFrame.
despesas_2021.drop('cod_municipio', axis=1, inplace=True)

In [21]:
despesas_2021.head()

seq_orgao  seq_unidade  cod_unidade  cod_subunidade  num_anoexercicio  \
0        372       390343         6001               4              2021   
1        372       390343         6001               4              2021   
2        372       390336         7001               3              2021   
3        372       390336         7001               3              2021   
4        372       390337         7001               2              2021   

   num_mesexercicio               dsc_funcao                    dsc_subfuncao  \
0                11  28 - ENCARGOS ESPECIAIS  843 - SERVIÇO DA DÍVIDA INTERNA   
1                 1  28 - ENCARGOS ESPECIAIS  843 - SERVIÇO DA DÍVIDA INTERNA   
2                 8            12 - EDUCAÇÃO          365 - EDUCAÇÃO INFANTIL   
3                 7            12 - EDUCAÇÃO          365 - EDUCAÇÃO INFANTIL   
4                 5            12 - EDUCAÇÃO         361 - ENSINO FUNDAMENTAL   

                dsc_programa  \
0  0000 - ENCARGOS ESPECIAIS   
1  0000 - ENCARGOS ESPECIAIS   
2     0024 - ENSINO INFANTIL   
3     0024 - ENSINO INFANTIL   
4  0020 - ENSINO FUNDAMENTAL   

                                            dsc_acao  ...  \
0  3020 - AMORTIZAÇÃO DE PARCELAMENTO DE DÍVIDAS ...  ...   
1  3020 - AMORTIZAÇÃO DE PARCELAMENTO DE DÍVIDAS ...  ...   
2            2051 - MANUTENÇÃO DO ENSINO PRÉ-ESCOLAR  ...   
3            2050 - MANUTENÇÃO DE CRECHES MUNICIPAIS  ...   
4            2044 - MANUTENÇÃO DO ENSINO FUNDAMENTAL  ...   

                                    dsc_fonterecurso vlr_previsto  \
0                          100 - RECURSOS ORDINÁRIOS         0.00   
1                          100 - RECURSOS ORDINÁRIOS    58,000.00   
2  101 - RECEITAS DE IMPOSTOS E DE TRANSFERÊNCIAS...         0.00   
3  101 - RECEITAS DE IMPOSTOS E DE TRANSFERÊNCIAS...         0.00   
4  101 - RECEITAS DE IMPOSTOS E DE TRANSFERÊNCIAS...         0.00   

  vlr_acrescimo  vlr_deducao  vlr_empenhado  vlr_liquidado  vlr_pago  \
0          0.00    58,000.00           0.00           0.00      0.00   
1          0.00         0.00           0.00           0.00      0.00   
2          0.00         0.00           0.00         155.11    134.70   
3          0.00         0.00           0.00         566.96      0.00   
4          0.00         0.00           0.00         444.39    598.26   

   vlr_rspprocessado  vlr_rspnprocessado  num_versao_arq  
0               0.00                0.00               1  
1               0.00                0.00               1  
2              20.41             -155.11               1  
3             566.96             -566.96               1  
4            -153.87             -444.39               1  

[5 rows x 22 columns]

In [22]:
# Excluindo colunas desnecessárias do dataframe.
# axis especifica se os dados serão descartados do índice/linha (0 ou index) ou coluna (1 ou column).
# inplace=True indica que os dados serão alterados diretamente no DataFrame.
pagamentos_2021.drop('seq_rsp', axis=1, inplace=True)

In [23]:
pagamentos_2021.head()

seq_pagamento  seq_orgao  seq_unidade  seq_empenho  seq_liquidacao  \
0      107016667        371       390308     67694508       114229168   
1      107016668        371       390307     67694479       114229181   
2      107016669        371       390308     67694498       110028021   
3      107016670        371       390308     70299539       114229153   
4      107016671        371       390307     67694471       114229188   

   seq_orgao_empenho  seq_unid_empenho  num_ano_referencia  \
0                371            390308                2021   
1                371            390307                2021   
2                371            390308                2021   
3                371            390308                2021   
4                371            390307                2021   

   num_mes_referencia num_doc_resp  ... num_ord_pagamento dat_pagamento  \
0                   2  080***586**  ...                43    15/02/2021   
1                   2  080***586**  ...                45    17/02/2021   
2                   2  080***586**  ...                25    05/02/2021   
3                   2  080***586**  ...                48    18/02/2021   
4                   2  080***586**  ...                59    26/02/2021   

                                       dsc_pagamento  \
0  PAGAMENTO QUE SE FAZ REFERENTE PAGAMENTO DE DE...   
1  PAGAMENTO QUE SE FAZ REFERENTE LOCAÇÃO E MONIT...   
2  PAGAMENTO QUE SE FAZ REFERENTE A AQUISIÇÃO DE ...   
3  PAGAMENTO QUE SE FAZ REFERENTE AO PAGAMENTO DE...   
4  REFERENTE A FOLHA DE PAGAMENTO DO MES DE FEVER...   

                                  dsc_tipo_pagamento  \
0  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
1  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
2  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
3  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
4  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   

           dsc_fonte_recurso  vlr_pag_fonte vlr_ret_fonte  vlr_ant_fonte  \
0  100 - RECURSOS ORDINÁRIOS          23.12          0.00           0.00   
1  100 - RECURSOS ORDINÁRIOS         395.00          0.00           0.00   
2  100 - RECURSOS ORDINÁRIOS          71.50          0.00           0.00   
3  100 - RECURSOS ORDINÁRIOS         600.00          0.00           0.00   
4  100 - RECURSOS ORDINÁRIOS      21,460.96          0.00           0.00   

  vlr_anu_fonte num_versao_arq  
0          0.00              1  
1          0.00              1  
2          0.00              1  
3          0.00              1  
4          0.00              1  

[5 rows x 27 columns]

## Convertendo o tipo dos valores

Após termos removidos do nosso DataFrame campos com valores irrelevantes para nossa análise; o próximo passo é analisar como os dados foram reconhecidos pelo _Pandas_ no momento da criação do DataFrame.

In [24]:
# Entendo os tipos dos dados para cada coluna.
despesas_2021.dtypes

seq_orgao                int64
seq_unidade              int64
cod_unidade              int64
cod_subunidade           int64
num_anoexercicio         int64
num_mesexercicio         int64
dsc_funcao              object
dsc_subfuncao           object
dsc_programa            object
dsc_acao                object
dsc_subacao            float64
dsc_naturezadespesa     object
dsc_fonterecurso        object
vlr_previsto           float64
vlr_acrescimo          float64
vlr_deducao            float64
vlr_empenhado          float64
vlr_liquidado          float64
vlr_pago               float64
vlr_rspprocessado      float64
vlr_rspnprocessado     float64
num_versao_arq           int64
dtype: object

In [25]:
pagamentos_2021.dtypes

seq_pagamento           int64
seq_orgao               int64
seq_unidade             int64
seq_empenho             int64
seq_liquidacao          int64
seq_orgao_empenho       int64
seq_unid_empenho        int64
num_ano_referencia      int64
num_mes_referencia      int64
num_doc_resp           object
nom_resp               object
num_doc_credor         object
nom_credor             object
num_empenho             int64
dat_empenho            object
num_liquidacao        float64
dat_liquidacao         object
num_ord_pagamento       int64
dat_pagamento          object
dsc_pagamento          object
dsc_tipo_pagamento     object
dsc_fonte_recurso      object
vlr_pag_fonte         float64
vlr_ret_fonte         float64
vlr_ant_fonte         float64
vlr_anu_fonte         float64
num_versao_arq          int64
dtype: object

O resultado da análise será tão precisa quando a fonte de dados for precisa, neste caso o Tribunal de Contas do Estado de Minas Gerais fez um trabalho incrível co criar esse dataset, praticamente não precisamos tratar os dados para realizar a análise, o _Pandas_ reconheceu os campos nos formatos esperados, só vamos retrabalhar os campos com as informações de datas para praticidade de uso.

In [26]:
# Convertendo a coluna dat_pagamento para valores do tipo datetime64 no formato DD/MM/YYYY.
pagamentos_2021.loc[:, 'dat_pagamento'] = pd.to_datetime(pagamentos_2021['dat_pagamento'], format='%d/%m/%Y')

In [27]:
pagamentos_2021.tail()

seq_pagamento  seq_orgao  seq_unidade  seq_empenho  seq_liquidacao  \
18378      118428698        372       390324     76166982       125584801   
18379      118428699        372       390322     74956657       125584811   
18380      118428700        372       390322     71953160       123477280   
18381      118428701        372       390324     76166874       123477444   
18382      118428702        372       390324     77450244       125584800   

       seq_orgao_empenho  seq_unid_empenho  num_ano_referencia  \
18378                372            390324                2021   
18379                372            390322                2021   
18380                372            390322                2021   
18381                372            390324                2021   
18382                372            390324                2021   

       num_mes_referencia num_doc_resp  ... num_ord_pagamento dat_pagamento  \
18378                   7  064***256**  ...              9568    2021-07-14   
18379                   7  064***256**  ...              9241    2021-07-14   
18380                   7  064***256**  ...              9208    2021-07-14   
18381                   7  064***256**  ...              9907    2021-07-23   
18382                   7  064***256**  ...              9855    2021-07-23   

                                           dsc_pagamento  \
18378  ORDEM DEPAGAMENTO  REFERENTE AQUISIÇÃO DE PROD...   
18379  ORDEM DE PAGAMENTO REFERENTE PRESTAÇÃO DE SERV...   
18380  OP REFERENTE AQUISIÇÃO DE MATERIAIS MÉDICO-HOS...   
18381  ORDEM DE PAGAMENTO REFERENTE A AQUISIÇÃO DE AD...   
18382  ORDEM DE PAGAMENTO REFERENTE A LOCAÇÃO DE SONO...   

                                      dsc_tipo_pagamento  \
18378  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
18379  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
18380  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
18381  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   
18382  1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍ...   

                                       dsc_fonte_recurso  vlr_pag_fonte  \
18378  155 - TRANSFERÊNCIAS DE RECURSOS DO FUNDO ESTA...         446.00   
18379  155 - TRANSFERÊNCIAS DE RECURSOS DO FUNDO ESTA...          63.35   
18380  155 - TRANSFERÊNCIAS DE RECURSOS DO FUNDO ESTA...          67.95   
18381  155 - TRANSFERÊNCIAS DE RECURSOS DO FUNDO ESTA...          64.00   
18382  155 - TRANSFERÊNCIAS DE RECURSOS DO FUNDO ESTA...         575.00   

      vlr_ret_fonte  vlr_ant_fonte vlr_anu_fonte num_versao_arq  
18378          0.00           0.00          0.00              1  
18379          0.00           0.00          0.00              1  
18380          0.00           0.00          0.00              1  
18381          0.00           0.00          0.00              1  
18382          0.00           0.00          0.00              1  

[5 rows x 27 columns]

# Análise dos Dados (Data Analysis)

Ate o momento já tivemos muito trabalho, porem só estávamos tratando os dados para que facilitar o processo de realizar cálculos matemáticos e criação de gráficos. Agora vamos tentar responder algumas perguntas utilizando esta base nos dados:


Começaremos identificando quais são os maiores gastos do município.

In [28]:
credores = pagamentos_2021.groupby('nom_credor')['vlr_pag_fonte'].sum().sort_values(ascending=False)

In [29]:
credores.head(10)

nom_credor
NAO INFORMADO                                                              33,487,890.36
INSTITUTO NACIONAL DO SEGURO SOCIAL INSS                                    8,701,279.30
FUNDACAO HOSPITALAR SAO VICENTE DE PAULO                                    3,282,717.51
CEMIG DISTRIBUIÇÃO S.A                                                      1,556,875.80
TRIBUNAL REGIONAL DO TRABALHO DA 03ª REGIAO VARA DO TRABALHO DE GUANHAES    1,417,316.34
CONSTRUTORA CIVIL FN LOURO LTDA ME                                          1,332,425.57
DECOR PRESENTES LTDA - ME                                                   1,218,153.25
AUTO POSTO NEVES LTDA                                                       1,179,436.34
POSITIVO CONSTRUTORA CIVIL LTDA                                             1,066,655.04
RECEITA FEDERAL DO BRASIL                                                     956,982.51
Name: vlr_pag_fonte, dtype: float64

_"Houston, we have a problem"_. Nosso principal credor não tem um nome definido, este é um comportamento estranho, visto que para a administração pública é obrigatório informar que é o destinatário de um pagamento. Vamos analisar de perto os lançamentos referente ao este credor `NAO INFORMADO`.

In [30]:
# Para a coluna nom_credor filtre apenas os campos que contenha o lançamento 'NAO INFORMADO'.
pagamentos_2021[pagamentos_2021['nom_credor'].str.contains('NAO INFORMADO', na=False)]['dsc_pagamento'].head()

777    OP REFERENTE A FOLHA DE PAGAMENTO DOS SERVIDOR...
792    OP REFERENTE A FOLHA DE PAGAMENTO DOS SERVIDOR...
799    OP REFERENTE A FOLHA DE PAGAMENTO DOS SERVIDOR...
805    OP REFERENTE A FOLHA DE PAGAMENTO DOS SERVIDOR...
814    OP REFERENTE A FOLHA DE PAGAMENTO DOS SERVIDOR...
Name: dsc_pagamento, dtype: object

In [31]:
pagamentos_2021[pagamentos_2021['nom_credor'].str.contains('NAO INFORMADO', na=False)]['dsc_pagamento'].head()[777]

'OP REFERENTE A FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS, RELATIVO AO MES DE NOVEMBRO DE 2021.'

Agora fez sentido, este pagamento é referente a quitação da folha de pagamento do município no decorrer do ano, portanto não é direcionando a um credor específico. Para fins de melhorar o processo de analise vamos renomear este lançamento para `FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS`.

In [32]:
# Trocando a informação 'NAO INFORMADO' da coluna nom_credor por FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS.
# inplace=True indica que os dados serão alterados diretamente no DataFrame.
pagamentos_2021['nom_credor'].replace('NAO INFORMADO', 'FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS', inplace=True)

Vamos totalizar os gastos por cada credor novamente.

In [33]:
credores = pagamentos_2021.groupby('nom_credor')[['vlr_pag_fonte', 'vlr_ret_fonte', 'vlr_ant_fonte', 'vlr_anu_fonte']].sum().sort_values(by='vlr_pag_fonte', ascending=False)

In [34]:
credores.head(10)

vlr_pag_fonte  \
nom_credor                                                          
FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS    33,487,890.36   
INSTITUTO NACIONAL DO SEGURO SOCIAL INSS             8,701,279.30   
FUNDACAO HOSPITALAR SAO VICENTE DE PAULO             3,282,717.51   
CEMIG DISTRIBUIÇÃO S.A                               1,556,875.80   
TRIBUNAL REGIONAL DO TRABALHO DA 03ª REGIAO VAR...   1,417,316.34   
CONSTRUTORA CIVIL FN LOURO LTDA ME                   1,332,425.57   
DECOR PRESENTES LTDA - ME                            1,218,153.25   
AUTO POSTO NEVES LTDA                                1,179,436.34   
POSITIVO CONSTRUTORA CIVIL LTDA                      1,066,655.04   
RECEITA FEDERAL DO BRASIL                              956,982.51   

                                                    vlr_ret_fonte  \
nom_credor                                                          
FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS     5,903,972.07   
INSTITUTO NACIONAL DO SEGURO SOCIAL INSS                     0.00   
FUNDACAO HOSPITALAR SAO VICENTE DE PAULO                     0.00   
CEMIG DISTRIBUIÇÃO S.A                                       0.00   
TRIBUNAL REGIONAL DO TRABALHO DA 03ª REGIAO VAR...           0.00   
CONSTRUTORA CIVIL FN LOURO LTDA ME                      83,276.58   
DECOR PRESENTES LTDA - ME                               13,278.00   
AUTO POSTO NEVES LTDA                                        0.00   
POSITIVO CONSTRUTORA CIVIL LTDA                          5,135.86   
RECEITA FEDERAL DO BRASIL                                    0.00   

                                                    vlr_ant_fonte  \
nom_credor                                                          
FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS             0.00   
INSTITUTO NACIONAL DO SEGURO SOCIAL INSS                     0.00   
FUNDACAO HOSPITALAR SAO VICENTE DE PAULO                     0.00   
CEMIG DISTRIBUIÇÃO S.A                                       0.00   
TRIBUNAL REGIONAL DO TRABALHO DA 03ª REGIAO VAR...           0.00   
CONSTRUTORA CIVIL FN LOURO LTDA ME                           0.00   
DECOR PRESENTES LTDA - ME                                    0.00   
AUTO POSTO NEVES LTDA                                        0.00   
POSITIVO CONSTRUTORA CIVIL LTDA                              0.00   
RECEITA FEDERAL DO BRASIL                                    0.00   

                                                    vlr_anu_fonte  
nom_credor                                                         
FOLHA DE PAGAMENTO DOS SERVIDORES(AS) MUNICIPAIS             0.00  
INSTITUTO NACIONAL DO SEGURO SOCIAL INSS                     0.00  
FUNDACAO HOSPITALAR SAO VICENTE DE PAULO                     0.00  
CEMIG DISTRIBUIÇÃO S.A                                       0.00  
TRIBUNAL REGIONAL DO TRABALHO DA 03ª REGIAO VAR...           0.00  
CONSTRUTORA CIVIL FN LOURO LTDA ME                           0.00  
DECOR PRESENTES LTDA - ME                                    0.00  
AUTO POSTO NEVES LTDA                                        0.00  
POSITIVO CONSTRUTORA CIVIL LTDA                              0.00  
RECEITA FEDERAL DO BRASIL                                    0.00

Quem são os principais ordenadores de gastos?

In [35]:
ordenadores_gastos = pagamentos_2021.groupby('nom_resp')[['vlr_pag_fonte', 'vlr_ret_fonte', 'vlr_ant_fonte', 'vlr_anu_fonte']].sum().sort_values(by='vlr_pag_fonte', ascending=False)

In [36]:
ordenadores_gastos

vlr_pag_fonte  vlr_ret_fonte  vlr_ant_fonte  \
nom_resp                                                                       
TADEU FILIPE FERNANDES DE ABREU  30,037,226.39   1,433,176.62           0.00   
CELIA PECANHA DE OLIVEIRA        26,519,836.63   2,411,458.34           0.00   
ANA PAULA MARTINS LOPES ARAUJO   18,841,952.56   2,577,986.34           0.00   
MARINE PIMENTA DE FIGUEIREDO      6,408,153.24     326,746.39           0.00   
GILMAR ISAIAS DOS SANTOS          2,277,278.24           0.00           0.00   
HENRIQUE CESAR SOARES BARBOSA         2,235.00           0.00           0.00   

                                 vlr_anu_fonte  
nom_resp                                        
TADEU FILIPE FERNANDES DE ABREU      20,847.13  
CELIA PECANHA DE OLIVEIRA            14,685.00  
ANA PAULA MARTINS LOPES ARAUJO            0.00  
MARINE PIMENTA DE FIGUEIREDO          8,754.00  
GILMAR ISAIAS DOS SANTOS                  0.00  
HENRIQUE CESAR SOARES BARBOSA             0.00

Pagamentos totalizados por tipo de recursos.

In [37]:
tipos_pagamento = pagamentos_2021.groupby('dsc_tipo_pagamento')[['vlr_pag_fonte', 'vlr_ret_fonte', 'vlr_ant_fonte', 'vlr_anu_fonte']].sum().sort_values(by='vlr_pag_fonte', ascending=False)

In [38]:
tipos_pagamento

vlr_pag_fonte  \
dsc_tipo_pagamento                                                  
1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍVIDA  80,954,141.48   
3 - PAGAMENTO DE RESTOS A PAGAR PROCESSADOS E N...   2,033,099.68   
2 - PAGAMENTO DE SERVIÇO DA DÍVIDA                     801,023.30   
4 - PAGAMENTO DE RESTOS A PAGAR NÃO PROCESSADOS...     298,417.60   

                                                    vlr_ret_fonte  \
dsc_tipo_pagamento                                                  
1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍVIDA   6,715,567.41   
3 - PAGAMENTO DE RESTOS A PAGAR PROCESSADOS E N...      18,795.95   
2 - PAGAMENTO DE SERVIÇO DA DÍVIDA                           0.00   
4 - PAGAMENTO DE RESTOS A PAGAR NÃO PROCESSADOS...      15,004.33   

                                                    vlr_ant_fonte  \
dsc_tipo_pagamento                                                  
1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍVIDA           0.00   
3 - PAGAMENTO DE RESTOS A PAGAR PROCESSADOS E N...           0.00   
2 - PAGAMENTO DE SERVIÇO DA DÍVIDA                           0.00   
4 - PAGAMENTO DE RESTOS A PAGAR NÃO PROCESSADOS...           0.00   

                                                    vlr_anu_fonte  
dsc_tipo_pagamento                                                 
1 - PAGAMENTO DE DESPESA, EXCETO SERVIÇO DA DÍVIDA      44,286.13  
3 - PAGAMENTO DE RESTOS A PAGAR PROCESSADOS E N...           0.00  
2 - PAGAMENTO DE SERVIÇO DA DÍVIDA                           0.00  
4 - PAGAMENTO DE RESTOS A PAGAR NÃO PROCESSADOS...           0.00

Pagamento totalizados por fonte de recursos.

In [39]:
fontes_recurso = pagamentos_2021.groupby('dsc_fonte_recurso')[['vlr_pag_fonte', 'vlr_ret_fonte', 'vlr_ant_fonte', 'vlr_anu_fonte']].sum().sort_values(by='vlr_pag_fonte', ascending=False)

In [40]:
fontes_recurso.head()

vlr_pag_fonte  \
dsc_fonte_recurso                                                   
100 - RECURSOS ORDINÁRIOS                           30,404,241.19   
102 - RECEITAS DE IMPOSTOS E DE TRANSFERÊNCIAS ...  13,539,188.36   
118 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO N...   9,532,545.03   
159 - TRANSFERÊNCIA DE RECURSOS DO SISTEMA ÚNIC...   8,084,014.63   
119 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO E...   3,652,085.69   

                                                    vlr_ret_fonte  \
dsc_fonte_recurso                                                   
100 - RECURSOS ORDINÁRIOS                            1,587,117.54   
102 - RECEITAS DE IMPOSTOS E DE TRANSFERÊNCIAS ...     980,894.59   
118 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO N...   1,715,026.70   
159 - TRANSFERÊNCIA DE RECURSOS DO SISTEMA ÚNIC...   1,313,776.51   
119 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO E...     531,214.31   

                                                    vlr_ant_fonte  \
dsc_fonte_recurso                                                   
100 - RECURSOS ORDINÁRIOS                                    0.00   
102 - RECEITAS DE IMPOSTOS E DE TRANSFERÊNCIAS ...           0.00   
118 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO N...           0.00   
159 - TRANSFERÊNCIA DE RECURSOS DO SISTEMA ÚNIC...           0.00   
119 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO E...           0.00   

                                                    vlr_anu_fonte  
dsc_fonte_recurso                                                  
100 - RECURSOS ORDINÁRIOS                               24,947.13  
102 - RECEITAS DE IMPOSTOS E DE TRANSFERÊNCIAS ...       2,100.00  
118 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO N...           0.00  
159 - TRANSFERÊNCIA DE RECURSOS DO SISTEMA ÚNIC...      12,585.00  
119 - TRANSFERÊNCIAS DO FUNDEB PARA APLICAÇÃO E...           0.00

Pagamentos totalizados ao longo dos meses.

In [41]:
import locale
# define a localidade POSIX para a configuração padrão do usuário.
locale.setlocale(locale.LC_ALL, '')

'Portuguese_Brazil.1252'

In [42]:
pagamentos_por_mes = pagamentos_2021.groupby(pagamentos_2021['dat_pagamento'].dt.strftime('%B'))[['vlr_pag_fonte', 'vlr_ret_fonte', 'vlr_ant_fonte', 'vlr_anu_fonte']].sum().sort_index()

# Categorizando e organizando os dados pelos meses do ano.
pagamentos_por_mes.index = pd.CategoricalIndex(pagamentos_por_mes.index, categories=['janeiro', 'fevereiro', 'março', 'abril', 'maio', 'junho', 'julho', 'agosto', 'setembro', 'outubro', 'novembro', 'dezembro'], ordered=True)
pagamentos_por_mes.sort_index(inplace=True)

In [43]:
pagamentos_por_mes

vlr_pag_fonte  vlr_ret_fonte  vlr_ant_fonte  vlr_anu_fonte
dat_pagamento                                                            
janeiro         3,147,091.73      32,119.77           0.00         300.00
fevereiro       6,276,919.60     533,351.62           0.00      18,692.00
março           9,135,955.86   1,044,785.88           0.00           0.00
abril           6,460,015.26     546,812.85           0.00           0.00
maio            3,968,428.68      64,367.99           0.00       2,900.13
junho           6,400,491.00     525,327.39           0.00           0.00
julho           9,597,862.05     996,929.55           0.00         500.00
agosto          4,261,003.25      59,269.51           0.00       1,900.00
setembro        7,178,244.68     648,037.88           0.00      16,055.00
outubro         9,697,209.47   1,217,436.09           0.00       2,089.00
novembro        5,097,982.77      95,323.53           0.00       1,450.00
dezembro       12,865,477.71     985,605.63           0.00         400.00